File: backend/src/main/java/com/whiteblue/backend/domain/application/controller/ApplicationController.java
package com.whiteblue.backend.domain.application.controller;

import com.whiteblue.backend.domain.application.service.ApplicationService;
import com.whiteblue.backend.domain.application.DTO.CreateApplicationDTO;
import com.whiteblue.backend.domain.application.entity.Application;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.security.oauth2.core.user.OAuth2User;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RequestMapping("/application")
@RestController
public class ApplicationController {
    private final ApplicationService applicationService;

    @PostMapping("/")
    public Application write(@Validated @RequestBody CreateApplicationDTO createApplicationDTO,
                             @AuthenticationPrincipal OAuth2User user) {
        return null;
    }
}

File: backend/src/main/kotlin/com/whiteblue/backend/domain/application/controller/ApplicationController.kt
package com.whiteblue.backend.domain.application.controller

import com.whiteblue.backend.domain.application.dto.ApplicationResponse
import com.whiteblue.backend.domain.application.dto.CreateApplicationRequest
import com.whiteblue.backend.domain.application.dto.UpdateApplicationRequest
import com.whiteblue.backend.domain.application.service.ApplicationServiceImpl
import com.whiteblue.backend.global.oAuth.OAuthUser
import jakarta.validation.constraints.Positive
import org.springframework.http.ResponseEntity
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*

@RequestMapping("/application")
@RestController
class ApplicationController(private val applicationService: ApplicationServiceImpl) {
    @GetMapping("")
    fun getMyApplication(@AuthenticationPrincipal oAuthUser: OAuthUser): ResponseEntity<ApplicationResponse?> =
        ResponseEntity.ok(
            applicationService.getApplicationByWriterId(oAuthUser.id)
        )

    @PostMapping("")
    fun createApplication(
        @AuthenticationPrincipal oAuthUser: OAuthUser,
        @RequestBody @Validated createApplicationRequest: CreateApplicationRequest,
    ): ResponseEntity<ApplicationResponse> =
        ResponseEntity.ok(
            applicationService.createApplication(oAuthUser.id, createApplicationRequest)
        )

    @PutMapping("/{id}")
    fun updateApplication(
        @PathVariable @Positive id: Long,
        @RequestBody @Validated updateApplicationRequest: UpdateApplicationRequest
    ): ResponseEntity<ApplicationResponse> =
        ResponseEntity.ok(
            applicationService.updateApplication(id, updateApplicationRequest)
        )

    @DeleteMapping("/{id}")
    fun deleteApplication(@PathVariable @Positive id: Long): ResponseEntity<Unit> =
        ResponseEntity.ok(
            applicationService.deleteApplication(id)
        )
}
